projects/SDE Lighthouse 04 - Bruce/template-sql/4.infections-hospital.template.sql

```diff
@@ -0,0 +1,48 @@
+--┌────────────────────────────────────┐
+--│ LH004 Infections file              │
+--└────────────────────────────────────┘
+
+-- From application:
+--	PneumococcalVaccination (ever recorded y/n) [5.vaccinations]
+--	PneumoVacDate [5.vaccinations]
+--	InfluenzaVaccination (in past 12 months y/n) [5.vaccinations]
+--	FluVacDate [5.vaccinations]
+--	COVIDVaccination (in past 12 months y/n) [5.vaccinations]
+--	COVIDVacDate [5.vaccinations]
+--	ShinglesVaccination [5.vaccinations]
+--	ShinglesVacDate [5.vaccinations]
+--	BCGVaccination [5.vaccinations]
+--	BCGVacDate [5.vaccinations]
+--	HPVVaccination [5.vaccinations]
+--	HPVVacDate [5.vaccinations]
+--	Infection (from list below) [4.infections-gp] [4.infections-hospital]
+--	InfectionDate (for each recorded infection) [4.infections-gp]
+--	HospitalAdmissionForInfection (if available) [4.infections-hospital]
+--	HospitalAdmissionDate [4.infections-hospital]
+--	PreviousSmear (ever y/n) [6.smears]
+--	SmearDates [6.smears]
+--	SmearResults (for each smear documented) [6.smears]
+-- Infections from : 
+-- https://data.bris.ac.uk/datasets/2954m5h0ync672u8yzx16xxj7l/infection_master_published.txt
+
+-- PI agreed to separate files for infections, vaccinations and smear tests
+
+--> CODESET infections:2 
+--> CODESET bone-infection:1 cardiovascular-infection:1 cellulitis:1 diverticulitis:1
+--> CODESET gastrointestinal-infections:1 genital-tract-infections:1 hepatobiliary-infection:1
+--> CODESET infection-other:1 lrti:1 muscle-infection:1 neurological-infection:1 peritonitis:1
+--> CODESET puerpural-infection:1 pyelonephritis:1 urti-bacterial:1 urti-viral:1 uti:2
+
+DROP TABLE IF EXISTS {{project-schema}}."LH004-4_infections_hospital";
+CREATE TABLE {{project-schema}}."LH004-4_infections_hospital" AS
+SELECT
+	SUBSTRING("Der_Pseudo_NHS_Number", 2)::INT AS "PatientID",
+	b.concept AS "Infection",
+	b.description AS "InfectionDescription",
+	"Admission_Date"
+FROM INTERMEDIATE.national_flows_apc."tbl_Data_SUS_APCE" a
+LEFT OUTER JOIN {{code-set-table}} b ON b.code = a."Der_Primary_Diagnosis_Code"
+where "APCS_First_Ep_Ind"='1'
```

Comment:
That's right - to avoid hospital acquired infections. Though given I ended up using the primary diagnosis code, rather than the list, I think I could have done first episode, or entire spell, and it would have been the same. Not sure though.